tests/test_radiation.py

```diff
@@ -0,0 +1,209 @@
+"""Test module for abiotic.radiation.py."""
+
+import numpy as np
+import pytest
+
+# from core.constants import CONSTANTS as C  # this doesn't exist yet
+from virtual_rainforest.models.abiotic.radiation import (
+    ALBEDO_SHORTWAVE,
+    ALBEDO_VIS,
+    BEER_REGRESSION,
+    BOLZMAN_CONSTANT,
+    CANOPY_EMISSIVITY,
+    CELSIUS_TO_KELVIN,
+    CLOUDY_TRANSMISSIVITY,
+    FLUX_TO_ENERGY,
+    SECOND_TO_DAY,
+    SOIL_EMISSIVITY,
+    TRANSMISSIVITY_COEFFICIENT,
+)
+
+
+def test_import_constants():
+    """Test that constants were imported correctly."""
+    assert CLOUDY_TRANSMISSIVITY == 0.25
+    assert TRANSMISSIVITY_COEFFICIENT == 0.50
+    assert FLUX_TO_ENERGY == 2.04
+    assert BOLZMAN_CONSTANT == 5.67e-8
+    assert SOIL_EMISSIVITY == 0.95
+    assert CANOPY_EMISSIVITY == 0.95
+    assert BEER_REGRESSION == 2.67e-5
+    assert ALBEDO_VIS == np.array(0.03, dtype=np.float32)
+    assert ALBEDO_SHORTWAVE == np.array(0.17, dtype=np.float32)
+    assert CELSIUS_TO_KELVIN == 273.15
+    assert SECOND_TO_DAY == 86400
```

Comment:
I'm not sure this is helpful. We would have to update it if the values changed and then you need to change a value in two places (module and here). Essentially, this is primarily a test that you typed the same number correctly twice.

Reply:
And why are two of the constants defined as `np.array`?

Reply:
The idea was to test that all constants were imported correctly.
Albedo is an array here because it is the default if there is no array for albedo provided as an input (albedo will be a function of land cover in the future).

Reply:
That makes sense, but I think the thing to test here is that the required constants _can_ be imported rather than their values. With the current test, it is possible that a constant might be mistyped in the code, but equally could be mistyped here. It's really a test of duplicated typing rather than the code. And - because the import is outside of the test - if one is missing, the test file itself will fail with an AttributeError. 

Does this do what you want? It checks that each constant is provided.

```python

@pytest.mark.parametrize(
    argnames=["cname"],
    argvalues=[
        ("ALBEDO_SHORTWAVE",),
        ("ALBEDO_VIS",),
        ("BEER_REGRESSION",),
        ("BOLZMAN_CONSTANT",),
        ("CANOPY_EMISSIVITY",),
        ("CELSIUS_TO_KELVIN",),
        ("CLOUDY_TRANSMISSIVITY",),
        ("FLUX_TO_ENERGY",),
        ("SECOND_TO_DAY",),
        ("SOIL_EMISSIVITY",),
        ("TRANSMISSIVITY_COEFFICIENT",),
    ],
)
def test_import_constants2(cname):
    """Test constants can be imported."""

    # Get the module that should contain the constants
    import virtual_rainforest.models.abiotic.radiation as rad
   
    # Check that the constant can be retrieved from the module without
    # raising an AttributeError.
    with does_not_raise():
        assert getattr(rad, cname)
```

Reply:
I think defaults for an argument should probably be in the function signature rather than the constants? That is a grey area though - my `pyrealm` package has a whole bunch of 'constants' tucked away in a general settings container which are really just experimentally derived values. I think the distinction is that I'd expect someone to use a constant set of those values across multiple models (unless they are explicitly testing sensitivity). But something like albedo might very well change across multiple models of  study site?

Reply:
It should be quite straight forward to use different albedo values, it does not need to be the default. I could just stick the default value in the function instead.